lib/zstd/src/thread.cpp

```diff
@@ -0,0 +1,66 @@
+#ifdef CONFIG_ZSTD_THREAD
+#include <zstd/thread.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(zstd_thread, CONFIG_ZSTD_LOG_LEVEL);
+
+namespace zstd
+{
+
+/*******************************************************************************
+ * Slab based new/delete?
+ ******************************************************************************/
+#ifdef CONFIG_ZSTD_NEW_SLAB
+
+static K_MEM_SLAB_DEFINE(thread_slab, sizeof(thread), \
+                  CONFIG_ZSTD_THREAD_NEW_SLAB_MAX_COUNT, 1);
+
+#endif
+
+/*******************************************************************************
+ * Thread stack pool
+ ******************************************************************************/
+/* Not sure why max isn't already defined? */
+#define max(a, b) ((a > b) ? a : b)
```

Comment:
Need to identify why this is happening. It only seems to happen when using the gnuarmemb toolchain; when using the ZephyrSDK this seems fine.